.github/workflows/crates-io.yml

```diff
@@ -14,6 +14,6 @@ jobs:
       - uses: actions/checkout@v5
       - uses: rust-lang/crates-io-auth-action@v1
         id: auth
-      - run: cargo publish
+      - run: cargo publish --allow-dirty
```

Comment:
Meh, I'm not really a fan of allowing a dirty repo during publish, _especially_ from CI where you don't know exactly what has changed. Here you're _assuming_ that the only thing that has changed is a lockfile update. 

But maybe in the future we'll forget about this line and make a change somewhere else in CI and essentially have non-reproducible publishing

Reply:
I haven't had this happen to me... I haven't used `cargo install --locked` in CI but I also didn't need to use `--allow-dirty`

Reply:
Interesting; I'm not sure why I haven't hit this then?

Reply:
Maybe but I don't think so... decently often I have outdated dependencies in my lockfile when I just haven't run `cargo update` in a while but it still seems to work 🤷‍♂️